lib/chai/core/assertions.js

```diff
@@ -1763,10 +1763,22 @@ module.exports = function (chai, _) {
       , isOwn = flag(this, 'own')
       , flagMsg = flag(this, 'message')
       , obj = flag(this, 'object')
-      , ssfi = flag(this, 'ssfi');
+      , ssfi = flag(this, 'ssfi')
+      , nameType = typeof name;
+
+    flagMsg = flagMsg ? flagMsg + ': ' : '';
+
+    if (isNested) {
+      if (nameType !== 'string') {
+        throw new AssertionError(flagMsg + 'the argument to `property` must be a string');
```

Comment:
What about telling that it must be string because the nested flag is on?

Something like:
```js
throw new AssertionError(flagMsg + 'the argument to `property` with nested flag must be a string');
```

I'm afraid that it will be confusing to sometimes say that it can be a `string`, `number` or `symbol` and sometimes just saying that it must be a `string`

Thoughts? @keithamus @solodynamo @meeber

Reply:
But rather than writing `nested flag` we should think something else as end user may not know about internal logic of `isNested`.  Suggestions ? @vieiralucas @meeber @keithamus

Reply:
Agreed, we can safely assume to get here they've written `.nested.propery()` so we can use that instead.

Reply:
A user could reach this error via the `assert` interface too, though, so `nested.property` wouldn't always be entirely accurate.

Reply:
I'm having a hard time thinking on a good message

```
the argument to `property` when nesting must be a `string`

the argument to `property` for a nested assertion must be a `string`
```

Reply:
How about:

**Normal:** the argument to `property` must be a `string`, `number`, or `symbol`
**Nested:** the argument to `property` must be a `string` when using `nested` syntax